src/source/Ice/Network.c

```diff
@@ -382,3 +383,68 @@ BOOL isSameIpAddress(PKvsIpAddress pAddr1, PKvsIpAddress pAddr2, BOOL checkPort)
 
     return ret;
 }
+
+#ifdef _WIN32
+INT32 getErrorCode(VOID)
+{
+    INT32 error = WSAGetLastError();
+    switch (error) {
+        case WSAEWOULDBLOCK:
+            error = EWOULDBLOCK;
+            break;
+        case WSAEINPROGRESS:
+            error = EINPROGRESS;
+            break;
+        case WSAEISCONN:
+            error = EISCONN;
+            break;
+        case WSAEINTR:
+            error = EINTR;
+            break;
+        default:
+            /* leave unchanged */
+            break;
+    }
+    return error;
+}
+#else
+INT32 getErrorCode(VOID)
+{
+    return errno;
+}
+#endif
+
+#ifdef _WIN32
+PCHAR getErrorString(INT32 error)
+{
+    static CHAR buffer[1024];
+    switch (error) {
+        case EWOULDBLOCK:
+            error = WSAEWOULDBLOCK;
+            break;
+        case EINPROGRESS:
+            error = WSAEINPROGRESS;
+            break;
+        case EISCONN:
+            error = WSAEISCONN;
+            break;
+        case EINTR:
+            error = WSAEINTR;
+            break;
+        default:
+            /* leave unchanged */
+            break;
+    }
+    if (FormatMessage((FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS), NULL, error, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), buffer,
+                      sizeof(buffer), NULL) == 0) {
```

Comment:
Could you please change sizeof() -> SIZEOF() and sprintf_s -> SNPRINTF

Reply:
Yes, I pushed another commit that makes this change.

I can squash the additional commit if you want me too.